File: app/src/main/java/me/imzack/app/ender/view/contract/MyPlansViewContract.java
package me.imzack.app.ender.view.contract;

import android.support.v7.widget.helper.ItemTouchHelper;

import me.imzack.app.ender.view.adapter.PlanListAdapter;
import me.imzack.app.ender.model.bean.Plan;

public interface MyPlansViewContract extends BaseViewContract {

    void showInitialView(PlanListAdapter planListAdapter, ItemTouchHelper itemTouchHelper, boolean isPlanItemEmpty);

    void onPlanItemClicked(int position);

    void onPlanCreated();

    void onPlanDeleted(Plan deletedPlan, int position, boolean shouldShowSnackbar);

    void onPlanItemEmptyStateChanged(boolean isEmpty);
}

File: app/src/main/java/net/zackzhang/app/end/view/contract/MyPlansViewContract.kt
package net.zackzhang.app.end.view.contract

import android.support.v7.widget.helper.ItemTouchHelper
import net.zackzhang.app.end.model.bean.Plan
import net.zackzhang.app.end.view.adapter.PlanListAdapter

interface MyPlansViewContract : BaseViewContract {

    fun showInitialView(planListAdapter: PlanListAdapter, itemTouchHelper: ItemTouchHelper, isPlanItemEmpty: Boolean)

    fun onPlanItemClicked(position: Int)

    fun onPlanCreated()

    fun onPlanDeleted(deletedPlan: Plan, position: Int, shouldShowSnackbar: Boolean)

    fun onPlanItemEmptyStateChanged(isEmpty: Boolean)
}
